pages/pipelines/glossary.md

```diff
@@ -0,0 +1,72 @@
+# Pipelines glossary
+
+The following terms describe key concepts to help you use Pipelines.
+
+{:toc}
+
+## Agent
+
+An agent is a small, reliable, and cross-platform build runner that connects your infrastructure to Buildkite. It polls Buildkite for work, runs jobs, and reports results. You can install agents on local machines, cloud servers, or other remote machines. You need at least one agent to run builds.
+
+To learn more, see the [Agent overview](/docs/agent/v3).
+
+## Artifact
+
+An artifact is a file or metadata generated during a build. You can keep artifacts in a Buildkite-managed storage service or a third-party cloud storage service like Amazon S3, Google Cloud Storage, or Artifactory. Common uses include storing assets like logs and reports, or passing files between steps.
+
+To learn more, see [Build artifacts](/docs/pipelines/artifacts).
+
+## Build
+
+A build is a single run of a pipeline. You can trigger a build in various ways, including through the UI, API, as the result of a webhook, or on a schedule.
+
+## Dynamic pipeline
+
+Dynamic pipelines define their steps at runtime using scripts, giving you the flexibility to only run the steps relevant to particular code changes and workflows.
+
+Dynamic pipelines are helpful when you have a complex build process that requires different steps to execute based on runtime conditions, such as the branch, the environment, or the results of previous steps.
+
+To learn more, see [Dynamic pipelines](/docs/pipelines/defining-steps#dynamic-pipelines).
+
+## Hook
+
+A hook is a method of customizing the behavior of Buildkite through lifecycle events. They let you run scripts at different points of the agent or job lifecycle. Using hooks, you can extend the functionality of Buildkite and automate tasks specific to your workflow and requirements.
+
+To learn more, see [Hooks](/docs/agent/v3/hooks).
+
+## Job
+
+A job is the execution of a command step during a build. Jobs run the commands, scripts, or plugins defined in the step.
+
+## Job queue
```

Comment:
I think we just call them "queues" most of the time. Is this enough? Is the "job" prefix necessary for disambiguation?

```
## Queue

Queues contain jobs waiting to run on an agent.
```

Reply:
I like it. I was nervous about whether we use other types of queue, but I don't think I've come across anything.